plugins/modules/zos_job_submit.py

```diff
@@ -628,25 +633,33 @@ def run_module():
     )
 
     module = AnsibleModule(argument_spec=module_args, supports_check_mode=True)
+    encoding = module.params.get('encoding')
+    if encoding is None:
+        encoding = {'from': DEFAULT_ASCII_CHARSET, 'to': DEFAULT_EBCDIC_CHARSET}
+    if encoding.get('from') is None:
+        encoding['from'] = DEFAULT_ASCII_CHARSET
+    if encoding.get('to') is None:
+        encoding['to'] = DEFAULT_EBCDIC_CHARSET
 
     arg_defs = dict(
         src=dict(arg_type="data_set_or_path", required=True),
         wait=dict(arg_type="bool", required=False),
         location=dict(
             arg_type="str", default="DATA_SET", choices=["DATA_SET", "USS", "LOCAL"],
         ),
-        encoding=dict(arg_type="encoding", default="UTF-8"),
```

Comment:
I understand why the old arg type would not work due to your new encoding parm structure, but would it be possible to add back in some sort of string checking? I know `quote()` should make things "safe" but having some sort of input sanitization/validation would probably be good

Reply:
For the BetterArgParser parsing, something like this could work (little hacky though):
```python
    arg_defs = dict(
        src=dict(arg_type="data_set_or_path", required=True),
        wait=dict(arg_type="bool", required=False),
        location=dict(
            arg_type="str", default="DATA_SET", choices=["DATA_SET", "USS", "LOCAL"],
        ),
        from_encoding=dict(arg_type="encoding", default=DEFAULT_ASCII_CHARSET),
        to_encoding=dict(arg_type="encoding", default=DEFAULT_EBCDIC_CHARSET),
        volume=dict(arg_type="volume", required=False),
        return_output=dict(arg_type="bool", default=True),
        wait_time_s=dict(arg_type="int", required=False, default=60),
        max_rc=dict(arg_type="int", required=False),
    )
   try:
        parser = BetterArgParser(arg_defs)
        parsed_args = parser.parse_args(module.params.update({"from_encoding": encoding.get('from'), "to_encoding": encoding.get("to"))
    except ValueError as err:
        module.fail_json(msg=str(err), **result)
```

Reply:
@blakeinate are you suggesting we sanitize all the charsets? Not sure i followed your suggestion, if so, it is something i thought we could do problem is if `iconv` changes the chartsets it supports we have to version track all these changes. I ended up adding this doc to support the charsets and added to the [encoding module doc here](https://github.com/ansible-collections/ibm_zos_core/blob/v1.1.0-beta1/plugins/modules/zos_encode.py#L108) and here is the doc site [link](https://ansible-collections.github.io/ibm_zos_core/supplementary.html#encode)

Reply:
@blakeinate Yeah, that is probably a good idea. My initial intent was that if the provided charsets are not valid, `iconv` would error out, which then can be relayed back to the user. But I agree that it can probably be handled earlier in the control flow.

Reply:
@ddimatos what was being done previously was a basic character check, essentially a regex containing all characters and min-max lengths of any existing encoding I could find, this ensures no characters that might be used for attempted shell injection could get through